common/djangoapps/config_models/models.py

```diff
@@ -35,33 +37,69 @@ def save(self, *args, **kwargs):
         Clear the cached value when saving a new configuration entry
         """
         super(ConfigurationModel, self).save(*args, **kwargs)
-        cache.delete(self.cache_key_name())
+        cache.delete(self.cache_key_name(*[getattr(self, key) for key in self.KEY_FIELDS]))
 
     @classmethod
-    def cache_key_name(cls):
+    def cache_key_name(cls, *args):
         """Return the name of the key to use to cache the current configuration"""
-        return 'configuration/{}/current'.format(cls.__name__)
+        if cls.KEY_FIELDS != ():
+            if len(args) != len(cls.KEY_FIELDS):
+                raise TypeError(
+                    "cache_key_name() takes exactly {} arguments ({} given)".format(len(cls.KEY_FIELDS), len(args))
+                )
+            return 'configuration/{}/current/{}'.format(
+                cls.__name__,
+                ','.join((arg if isinstance(arg, basestring) else repr(arg)) for arg in args)
```

Comment:
A few weird exceptions aside, for builtin types `repr()` and `str()` have more or less the same result, except for strings; `repr('3')` will include the quotes, while `str('3')` won't.  If you use `repr()` for everything but strings, and `str()` for strings, you might just as well use `str()` for all of them.